### PR TITLE
Adding facet on the main climatic station dialog and increase the number of shapes

### PR DESCRIPTION
**Pull request by jackykemboi**
@dannyparsons  
- I have added facets on the main climatic dialog.
- I have also make the fill optional to allow one produce maps like the one below which are not clear with the fill.
image

- Will it be a good idea to have tooltip on the shape to inform the user that you only use shape when the factor variable has less than six levels.

### CODE REVIEW

**Comment by rdstern** [comment]
I looked here on google and wonder if there is a way to have more.  It mentions the [scale_shape_manual](https://stackoverflow.com/questions/26223857/more-than-six-shapes-in-ggplot). 

Here is another [place](https://stackoverflow.com/questions/16813278/cycling-through-point-shapes-when-more-than-6-factor-levels).

Don't spend too long on this point, but finding a problem of this type and searching and finding a reasonably elegant solution would be a good item to include also when the work is written up.  I would be happy that there is a limit and it could be reasonably low  -say 10 or 12 or 15, but it would be nice if it were a bit higher than 6.  Also even nicer if you could extend the limit manually (through the script) if you wanted to change it - though that's not necessary.

This might also be a useful problem to discuss in our graphics as a whole and that would also be good to write up.  (I ma just about to query if one of Wycliffe, Yvonne, Evans would like to take up the challenge of improving the graphics, supervised by Maxwell.  Maybe they could work a bit with you as well?

**Comment by jackykemboi** [comment]
@africanmathsinitiative/developers  This is ready for testing

File: instat/dlgClimaticStationMaps.vb [hunk 1]
```diff
@@ -107,15 +109,18 @@ Public Class dlgClimaticStationMaps
...
         clsGGplotOperator = New ROperator
         clsXlimFunction = New RFunction
         clsYlimFunction = New RFunction
+        clsFacetOp = New ROperator
 
         ucrSelectorOutline.Reset()
         ucrReceiverFill.SetMeAsReceiver()
         ucrSelectorStation.Reset()
         ucrReceiverLongitude.SetMeAsReceiver()
+        ucrReceiverFacet.SetMeAsReceiver()
```
Comment (Muthenya) [inline instat/dlgClimaticStationMaps.vb]: There is more than one receiver that is set as the default. Only one needs to be set as the default receiver.

File: instat/dlgClimaticStationMaps.vb [hunk 2]
```diff
@@ -14,12 +14,13 @@
 ' You should have received a copy of the GNU General Public License 
 ' along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+Imports instat
```
Comment (Muthenya) [inline instat/dlgClimaticStationMaps.vb]: Please remove this.

File: instat/dlgClimaticStationMaps.vb [hunk 3]
```diff
@@ -214,7 +214,7 @@ Public Class dlgClimaticStationMaps
         ElseIf ucrReceiverLongitude.IsEmpty AndAlso Not ucrReceiverLatitude.IsEmpty Then
             bOkEnabled = False
         ElseIf Not ucrReceiverLongitude.IsEmpty AndAlso Not ucrReceiverLatitude.IsEmpty AndAlso ucrSaveMap.IsComplete Then
```
Comment (Muthenya) [inline instat/dlgClimaticStationMaps.vb]: I think this can be done differently.

**Comment by jackykemboi** [comment]
@Muthenya I have resolved the issues. Kindly check once again.

**Comment by Muthenya** [comment]
This is now okay.

**Comment by jackykemboi** [comment]
@dannyparsons  This is ready for you

**Comment by Patowhiz** [comment]
@jackykemboi how do I access this dialog. I would also like to go through it.

**Comment by jackykemboi** [comment]
@Patowhiz  Pull the StationMapsEdit branch. Go to climate>check Data>Station Maps the climaticStation dialog will appear.

**Comment by dannyparsons** [comment]
@Patowhiz great if you can test this before I look at it.

**Comment by Muthenya** [comment]
I have taken up this. I'll be opening a new pull request on this.

**Comment by maxwellfundi** [comment]
@Muthenya  this can be closed now?

**Comment by Muthenya** [comment]
Yes, as I have opened another pull request to continue in the work.

**Pull request closed without merging**